SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialPackageMapClient.cpp

```diff
@@ -163,18 +187,29 @@ FNetworkGUID FSpatialNetGUIDCache::AssignNewEntityActorNetGUID(AActor* Actor, co
 		uint32 Offset = Pair.Value;
 
 		FNetworkGUID SubobjectNetGUID = GetOrAssignNetGUID_SpatialGDK(Subobject);
-		FUnrealObjectRef SubobjectRef(EntityId, Offset);
-		RegisterObjectRef(SubobjectNetGUID, SubobjectRef);
+		FUnrealObjectRef EntityIdSubobjectRef(EntityId, Offset);
+
+		if (Subobject->IsNameStableForNetworking())
+		{
+			// Startup Actors have two valid UnrealObjectRefs: the entity id and the path.
+			// AssignNewStablyNamedObjectNetGUID will register the path ref.
+			FNetworkGUID SubobjectNetGUID = GetOrAssignNetGUID_SpatialGDK(Subobject);
+			FUnrealObjectRef StablyNamedSubobjectRef(0, 0, Subobject->GetFName().ToString(), StablyNamedRef);
+
+			UnrealObjectRefToNetGUID.Emplace(StablyNamedSubobjectRef, SubobjectNetGUID);
```

Comment:
If the subobject was referenced before, there would be an entry in `NetGUIDToUnrealObjectRef` that we need to replace.